### PR TITLE
Tenant creation closes #10

### LINKED ISSUE #10: As an unauthenticated user, when I create an account, I can request that my account become a borrower

### PR DESCRIPTION
**Pull request by mikedao**
We did a LOT here.

The original plan was to have a person create a user, and with a checkbox, allow that person to then fill out some stuff to create an organization/tenant. The problem is that because we cannot create a Tenant from a User because a User belongs to a Tenant, I had to create a new user flow where a user has to first choose if they are a borrower or a lender. If they are a borrower, they have to create an organization and then create a user. To give the user the tenant_id, I shoved the tenant Id that was just created into the session, and pulled it from there on creation of the user. This was deemed the best choice of action as putting user creation logic in the tenant controller seemed like the WORST of ideas. 

List of major changes"

* Created a static pages controller and supporting files to have a static page where the user could decide if they are a borrower or a lender. This whole process may be streamlined with the use of Javascript once we learn how to do it.

* Added password confirmation to Users controller strong params method. Some reason, it wasn't in there earlier.

* Updated the nav bar so sign up goes to the static page.

* Also have a migration removing index from users. Not rolling it back because I don't think it hurts anything.

* Also updated a lot of the tests to be true feature/integration tests.

### CODE REVIEW

**Comment by AdamSmith910** [comment]
Looks great.  I don't know how to put any sparklies in so I'm just going to merge.

File: app/controllers/static_pages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class StaticPagesController < ApplicationController
```
Comment (EmilyMB) [inline app/controllers/static_pages_controller.rb]: :heart: the use of a static pages controller

File: app/controllers/tenants_controller.rb [hunk 2]
```diff
@@ -4,4 +4,31 @@ def index
 
   def show
   end
+
+  def new
+  end
+
+  def create
```
Comment (EmilyMB) [inline app/controllers/tenants_controller.rb]: Haha. So, maybe we should have a waffle ticket to refactor this? Almost like the Gilded Rose Kata...jk, not nearly that bad :smile:

Comment (mikedao) [inline app/controllers/tenants_controller.rb]: How would you suggest we pull some of that logic out?

File: app/helpers/static_pages_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (EmilyMB) [inline app/helpers/static_pages_helper.rb]: Maybe delete this? We can always re-create if we need it.

File: app/views/layouts/_nav_bar.html.erb [hunk 4]
```diff
@@ -16,7 +16,7 @@
     <%= menu_item "Cart", showcart_path %>
     <%= menu_item "Admin Dashboard", admin_dashboard_path if current_user && current_user.admin? %>
     <% if current_user.nil? %>
-    	<%= menu_item "Signup", new_user_path %>
+    	<%= menu_item "Signup", choose_path %>
```
Comment (EmilyMB) [inline app/views/layouts/_nav_bar.html.erb]: This seems not RESTful...maybe there is some naming convention for switch methods? Choose_path just seems a little odd to me.

File: config/routes.rb [hunk 5]
```diff
@@ -4,10 +4,12 @@
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy", as: "logout"
 
+  resources :tenants, only: [:new, :create]
```
Comment (EmilyMB) [inline config/routes.rb]: :+1: for using only: !

File: test/integration/user_signup_test.rb [hunk 6]
```diff
@@ -32,4 +36,76 @@ class CreateAUserTest < ActionDispatch::IntegrationTest
...
+    fill_in "signup[country]", with: "country"
+    fill_in "signup[password]", with: "password"
+    fill_in "signup[password_confirmation]", with: "password"
+    fill_in "signup[email]", with: "email@example.com"
+    click_link_or_button "Create Account"
+
+    assert_equal root_path, current_path
+    assert page.has_content?("Thank you for creating an account.")
+  end
 end
```
Comment (EmilyMB) [inline test/integration/user_signup_test.rb]: :100: on all these tests!

**Comment by EmilyMB** [comment]
This looks pretty good :bee: . I'm still wondering if there could be a better way to do the user/borrower signup, but I guess we can look into that later if we have time. As noted, it might be worthwhile to open a waffle ticket to remember to come back and refactor the long methods.

**Pull request merged**